etest/build.py

```diff
@@ -1,32 +1,105 @@
 """Build command for etest Docker images."""
-import os
-import sys
+from enum import Enum
 from pathlib import Path
-from typing import List
 
 import click
 
+from etest import docker, qemu
+from etest.profile import Profile
+
+
+class __commands(Enum):
+    glibc = """
+/bin/bash -c \
+'emerge-webrsync && \
+echo en_US.UTF8 UTF-8 >> /etc/locale.gen && \
+echo en_US ISO-8859-1 >> /etc/locale.gen && \
+locale-gen && \
+eselect locale set en_US.utf8'
+    """
+
+    musl = """
+bin/bash -c \
+'touch /etc/portage/repos.conf/musl.conf && \
+echo \
+"[musl]
+location = /var/db/repos/musl
+sync-type = git
+sync-uri = https://github.com/gentoo-mirror/musl.git" >> /etc/portage/repos.conf/musl.conf && \
+emerge-webrsync && \
+emerge dev-vcs/git && \
+emerge --sync musl'
+    """
+
+    uclibc = "/bin/bash -c emerge-webrsync"
+
 
 @click.command(name="etest-build")
+@click.option("-q", "--quiet", is_flag=True, default=False, help="Suppress all output.")
+@click.option("-v", "--verbose", is_flag=True, default=False, help="Provide more output.")
 @click.option("--hardened/--no-hardened", default=False, help="Use a hardened profile.")
 @click.option("--multilib/--no-multilib", default=True, help="Use a multilib profile.")
 @click.option("--systemd/--no-systemd", default=False, help="Use a systemd profile.")
-# The following could and should be excised to an architecture module that
-# handles the nuances of architecture differences in Gentoo and Docker.
 @click.option(
     "--architecture",
     "--arch",
-    type=click.Choice(["amd64", "x86", "arm64", "armv5", "armv7", "ppc64"], case_sensitive=False),
+    type=click.Choice(["amd64", "x86", "arm64", "armv5", "armv6", "armv7", "ppc64"], case_sensitive=False),
     default="amd64",
     help="Architecture for the built image.",
 )
-@click.option("--environment", "--env", multiple=True, help="Pass an environment variable through to Docker.")
-@click.option("--name", type=Path, default=os.path.basename(sys.argv[0]), help="Name of the directory?")
-def main(hardened: bool, multilib: bool, systemd: bool, architecture: str, environment: List[str], name: Path) -> None:
-    """Build the etest image."""
-    click.echo(f"hardened: {hardened}")
-    click.echo(f"multilib: {multilib}")
-    click.echo(f"systemd: {systemd}")
-    click.echo(f"architecture: {architecture}")
-    click.echo(f"environment: {environment}")
-    click.echo(f"name: {name}")
+@click.option(
+    "--libc",
+    type=click.Choice(["glibc", "musl", "uclibc"], case_sensitive=False),
+    default="glibc",
+    help="libc for the built image.",
+)
+@click.option(
+    "--path",
+    type=click.Path(exists=True),
+    default=str(Path.cwd() / "Dockerfile"),
+    help="Path to the Dockerfile.",
+)
+def main(
+    quiet: bool,
+    verbose: bool,
+    hardened: bool,
+    multilib: bool,
+    systemd: bool,
+    architecture: str,
+    libc: str,
+    path: str,
+) -> None:
+    """Build the etest image/s."""
+    path_ = Path(path)
```

Comment:
This is only used once.  Just inline the cast at that location.

Reply:
I thought you wanted everything to have high-level types. Inlining is ok tho, alltough it will go from str to Path to str